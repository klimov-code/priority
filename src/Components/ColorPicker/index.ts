export { ColorPicker } from './ColorPicker';
export { Container, Circle } from './styled';
