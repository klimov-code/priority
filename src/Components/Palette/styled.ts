import styled, { css } from 'styled-components';

const Container = styled.div`
  padding: 22px 11px 11px;

  border-radius: 5px;
  background: white;
  width: 292px;

  box-sizing: border-box;
  box-shadow: 0 22px 70px 0 rgba(0, 0, 0, 0.08);
`;

const Title = styled.span`
  display: block;
  font-size: 13px;
  line-height: 18px;
  color: #454c54;

  margin-bottom: 20px;
  margin-left: 11px;
`;

const Colors = styled.div`
  display: flex;
  justify-content: space-between;
  flex-wrap: wrap;
`;

const Border = css`
  box-shadow: inset 0 0 0 2px #8cc7f2;
`;

const Color = styled.div`
  width: 32px;
  height: 32px;
  background: #454c54;
  border-radius: 50%;
  cursor: pointer;

  margin: 11px;

  ${({ isActive }: { isActive: boolean }) => isActive && Border};
`;

export { Title, Container, Colors, Color };
