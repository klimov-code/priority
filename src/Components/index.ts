export { Priority } from './Priority';
export { Editor } from './Editor';
export { Viewer } from './Viewer';
export { Header } from './Header';
