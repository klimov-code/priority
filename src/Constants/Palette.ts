const REDHUE = ['#ff2534', '#ff5e76', '#fd89c3', '#f326c0', '#c70056'];
const YELLOWHUE = ['#ffc100', '#ffd919', '#fccb86', '#ff9800', '#ff6536'];
const GREENHUE = ['#66cc88', '#97e053', '#ccdd1e', '#46af4a', '#009687'];
const BLUEHUE = ['#1a8fe6', '#30beff', '#62dff5', '#00bbd5', '#1043a7'];
const PURPLEHUE = ['#7531e4', '#d46ffc', '#dab0ff', '#9b3aee', '#9c1ab1'];
const BLACKHUE = ['#6f757c', '#9d9d9d', '#d1d1d1', '#4a4a4a', '#141414'];

const PALETTE = [...REDHUE, ...YELLOWHUE, ...GREENHUE, ...BLUEHUE, ...PURPLEHUE, ...BLACKHUE];

export { PALETTE };
