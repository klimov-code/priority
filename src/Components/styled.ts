import styled from 'styled-components';

const Text = styled.span`
  display: block;
  font-size: 13px;
  line-height: 18px;
  color: #454c54;
`;

const TextBold = styled(Text)`
  font-weight: bold;
`;

export { Text, TextBold };
