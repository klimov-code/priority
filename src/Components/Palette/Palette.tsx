import React from 'react';

import { PALETTE } from '~/Constants';

import { Container, Title, Colors, Color } from './styled';

const Palette: React.FC<{
  setColor: (color: string) => void;
  activeColor: string;
}> = ({ setColor, activeColor }) => (
  <Container>
    <Title>{'Select color'}</Title>
    <Colors>
      {PALETTE.map((color, index) => (
        <Color
          style={{ background: color }}
          key={index}
          isActive={color === activeColor}
          color={color}
          onClick={() => setColor(color)}
        />
      ))}
    </Colors>
  </Container>
);

export { Palette };
