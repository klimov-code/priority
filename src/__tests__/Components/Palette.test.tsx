import React from 'react';
import { render } from 'react-testing-library';
import { Palette } from '~/Components/Palette';

describe('Palette', () => {
  it('renders correctly', () => {
    const setColor = jest.fn();
    const activeColor = '#4a4a4a';

    const { container } = render(<Palette setColor={setColor} activeColor={activeColor} />);

    expect(container).toMatchSnapshot();
  });
});
