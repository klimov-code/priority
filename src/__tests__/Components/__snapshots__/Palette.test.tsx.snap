// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Palette renders correctly 1`] = `
<div>
  <div
    class="sc-bdVaJa jNHbSE"
  >
    <span
      class="sc-bwzfXH kmvPZI"
    >
      Select color
    </span>
    <div
      class="sc-htpNat hYLKwx"
    >
      <div
        class="sc-bxivhb htFudt"
        style="background: rgb(255, 37, 52);"
      />
      <div
        class="sc-bxivhb htFudt"
        style="background: rgb(255, 94, 118);"
      />
      <div
        class="sc-bxivhb htFudt"
        style="background: rgb(253, 137, 195);"
      />
      <div
        class="sc-bxivhb htFudt"
        style="background: rgb(243, 38, 192);"
      />
      <div
        class="sc-bxivhb htFudt"
        style="background: rgb(199, 0, 86);"
      />
      <div
        class="sc-bxivhb htFudt"
        style="background: rgb(255, 193, 0);"
      />
      <div
        class="sc-bxivhb htFudt"
        style="background: rgb(255, 217, 25);"
      />
      <div
        class="sc-bxivhb htFudt"
        style="background: rgb(252, 203, 134);"
      />
      <div
        class="sc-bxivhb htFudt"
        style="background: rgb(255, 152, 0);"
      />
      <div
        class="sc-bxivhb htFudt"
        style="background: rgb(255, 101, 54);"
      />
      <div
        class="sc-bxivhb htFudt"
        style="background: rgb(102, 204, 136);"
      />
      <div
        class="sc-bxivhb htFudt"
        style="background: rgb(151, 224, 83);"
      />
      <div
        class="sc-bxivhb htFudt"
        style="background: rgb(204, 221, 30);"
      />
      <div
        class="sc-bxivhb htFudt"
        style="background: rgb(70, 175, 74);"
      />
      <div
        class="sc-bxivhb htFudt"
        style="background: rgb(0, 150, 135);"
      />
      <div
        class="sc-bxivhb htFudt"
        style="background: rgb(26, 143, 230);"
      />
      <div
        class="sc-bxivhb htFudt"
        style="background: rgb(48, 190, 255);"
      />
      <div
        class="sc-bxivhb htFudt"
        style="background: rgb(98, 223, 245);"
      />
      <div
        class="sc-bxivhb htFudt"
        style="background: rgb(0, 187, 213);"
      />
      <div
        class="sc-bxivhb htFudt"
        style="background: rgb(16, 67, 167);"
      />
      <div
        class="sc-bxivhb htFudt"
        style="background: rgb(117, 49, 228);"
      />
      <div
        class="sc-bxivhb htFudt"
        style="background: rgb(212, 111, 252);"
      />
      <div
        class="sc-bxivhb htFudt"
        style="background: rgb(218, 176, 255);"
      />
      <div
        class="sc-bxivhb htFudt"
        style="background: rgb(155, 58, 238);"
      />
      <div
        class="sc-bxivhb htFudt"
        style="background: rgb(156, 26, 177);"
      />
      <div
        class="sc-bxivhb htFudt"
        style="background: rgb(111, 117, 124);"
      />
      <div
        class="sc-bxivhb htFudt"
        style="background: rgb(157, 157, 157);"
      />
      <div
        class="sc-bxivhb htFudt"
        style="background: rgb(209, 209, 209);"
      />
      <div
        class="sc-bxivhb jIedBS"
        style="background: rgb(74, 74, 74);"
      />
      <div
        class="sc-bxivhb htFudt"
        style="background: rgb(20, 20, 20);"
      />
    </div>
  </div>
</div>
`;
